src/Components/Compute/AuctionBlock/Auction.js

```diff
@@ -0,0 +1,76 @@
+import React from 'react';
+
+import AuctionBlock from '../../UI/AuctionBlock/AuctionBlock';
+
+export default class Auction extends React.Component {
+  constructor(props) {
+    super(props);
+    this.initialState = {
+      biddingSequence: props.biddingSequence,
+      highestBid: null,
+      winningPlayer: null,
+      currentBid: null,
+    };
+    this.state = { ...this.initialState };
+  }
+
+  componentDidUpdate(prevProps) {
+    const { biddingSequence, name } = this.props;
+    if (prevProps.name !== name) {
+      this.setState({ biddingSequence });
```

Comment:
`getDrivedStateFromProps` is better alternative here. It will save you one additional render.